web/src/components/data-fields/address/UnitesStatesAddress.tsx

```diff
@@ -13,6 +13,7 @@ interface Props {
   onValidation: () => void;
   isFullWidth?: boolean;
   excludeNJ?: boolean | true;
+  required?: boolean | false;
```

Comment:
your using 1 prop to drive the required behavior of most of the fields, this is not a good approach because you are coupling them and address line 2 is not using this prop. The prop should be broken up so the fields can be flexible. It's a bit confusing.